### ISSUE TITLE
Another native compilation broken by adding `"queryAllDeclaredMethods": true`

### ISSUE THREAD

**Issue by sdeleuze** [oracle/graal#6529]
Similar to #6510 but not applied on record, we see another regression when adding `"queryAllDeclaredMethods": true` on Spring beans which breaks any project using Spring HATEOAS, see https://github.com/spring-projects/spring-framework/issues/30407 related issue on Spring side.

I tried with [latest GraalVM 23.1.0-dev build](https://github.com/graalvm/graalvm-ce-dev-builds/releases/tag/23.1.0-dev-20230428_2141) but I get this error:
```
> com.oracle.svm.core.util.VMError$HostedError: com.oracle.svm.core.util.UserError$UserException: Image heap writing found a class not seen during static analysis. Did a static field or an object referenced from a static field change during native image generation? For example, a lazily initialized cache could have been initialized during image generation, in which case you need to force eager initialization of the cache before static analysis or reset the cache using a field value recomputation.

    class: [Lorg.springframework.plugin.core.Plugin;
  reachable through:
    object: [Ljava.lang.Class;@6564dc2d  of class: java.lang.Class[]
    object: com.oracle.svm.core.code.ImageCodeInfo@4c1fed69  of class: com.oracle.svm.core.code.ImageCodeInfo
    root: com.oracle.svm.core.code.ImageCodeInfo.prepareCodeInfo()
```
I have attached the report [svm_err_b_20230502T095049.941_pid19874.md](https://github.com/oracle/graal/files/11371019/svm_err_b_20230502T095049.941_pid19874.md).

The reproducer is https://github.com/eiswind/mre-hateoas-breaks-native. It can be workaround with a `org.springframework.plugin.core.Plugin[]` reflection hint but we would like a proper fix if possible backported to GraalVM 22.3.x maintenance release and GraalVM 23.0.

**Comment by sdeleuze** [comment]
Like the workaround we have added in #6510 in a generic way for record, I would be interested to know what pattern triggers this issue to choose on Spring side between 2 workaround:
 - Implement a generic workaround at Spring Framework level if the pattern here is a common one
 - Implement a specific workaround at Spring HATEOAS or Plugin level (adding `org.springframework.plugin.core.Plugin[]` reflection hint) if this is a unusual / specific arrangement

**Comment by cstancu** [comment]
This is caused by a bug in the [reflection metadata registration for generic arrays for `OrderAwarePluginRegistry.of(T[])`](https://github.com/spring-projects/spring-plugin/blob/main/core/src/main/java/org/springframework/plugin/core/OrderAwarePluginRegistry.java#L100). @loicottet is working on a fix.

**Comment by loicottet** [comment]
https://github.com/oracle/graal/pull/6547 is the PR for the fix. @sdeleuze can you check if this indeed solves the problem?

**Comment by sdeleuze** [comment]
@loicottet I built a custom `native-image` version from this PR sources and confirm it fixes the issue.

Can the fix be backported in GraalVM 22.3.x maintenance release and GraalVM upcoming June release?

**Comment by loicottet** [comment]
Yes, this will be included in both 23.0 and 22.3 updates.
